spyder/plugins/profiler/widgets/main_widget.py

```diff
@@ -560,10 +560,7 @@ def start(self, wdir=None, args=None, pythonpath=None):
         if args:
             p_args.extend(shell_split(args))
 
-        executable = sys.executable
-        if executable.endswith("spyder.exe"):
-            # py2exe distribution
-            executable = "python.exe"
+        executable = self.get_conf('executable', section='main_interpreter')

```

Comment:
I see two options about this change: you could remove it and leave it for PR #17789, or leave it here and close that PR (since it's so small). It's up to you.

Reply:
I will leave it here and close the other PR